libmattermost.c

```diff
@@ -3467,6 +3467,28 @@ mm_get_users_of_room(MattermostAccount *ma, MattermostChannel *channel)
 	g_free(url);
 }
 
+static gint64
+mm_get_channel_approximate_view_time(MattermostAccount *ma, MattermostChannel *channel)
+{
+	const gchar *tmptime = NULL;
+
+	PurpleChat *chat = mm_purple_blist_find_chat(ma, channel->id);
+	if (chat) {
+		tmptime = purple_blist_node_get_string(PURPLE_BLIST_NODE(chat),"channel_approximate_view_time");
+	} else {
+		PurpleBuddy *buddy = purple_blist_find_buddy(ma->account,g_hash_table_lookup(ma->one_to_ones,channel->id));
+		if (buddy) {
+			tmptime = purple_blist_node_get_string(PURPLE_BLIST_NODE(buddy),"channel_approximate_view_time");
+		}
+	}
+
+	if(!tmptime) {
+		tmptime = g_strdup_printf("%" G_GINT64_FORMAT, (g_get_real_time() / 1000)); // now.

```

Comment:
This will leak memory, needs to be `g_free`d (but not the other `tmptime`'s above)

Reply:
Oh, I just saw that the code that you moved out into it's own function had this same memleak

Reply:
Ok, the easiest way is to just strdup the strings from purple_blist_get_node_string and free it afterwards.